hera_stats/jackknives.py

```diff
@@ -0,0 +1,353 @@
+            boot = allspecs[inds]
+            nsamps = nsamples[inds]
+            ints = integs[inds]
+
+            boot_uvp = copy.deepcopy(uvp)
+
+            # Replace old arrays with new ones.
+            boot_uvp.data_array[spw] = boot
+            boot_uvp.nsample_array[spw] = nsamps
+            boot_uvp.integration_array[spw] = ints
+
+            # Average spectra
+            boot_uvp.average_spectra([blpairs], time_avg=True)
+
+            # Get data and append
+            boot_spec = boot_uvp.get_data((spw, uvp.get_blpairs()[0], pol))[0]
+            boot_l.append(boot_spec)
+
+        # Calculate error bar and save to stats_array
+        boot_l = np.vstack(boot_l)
+        errs = np.std(boot_l.real, axis=0) + 1j * np.std(boot_l.imag, axis=0)
+        uvp_avg.set_stats("bootstrap_errs", (spw, uvp.get_blpairs()[0], pol), errs[None])
+
+    return uvp_avg
+
+def bootstrap_jackknife(uvpl, pol="xx", n_boots=50):
+    """
+    Bootstraps a list of jackknife pairs.
+
+    Parameters
+    ----------
+    uvpl: list of UVPSpec pairs
+        The jackknived data, with UVPSpec.labels as the group and UVPSpec.jkftype as 
+        the jackknife type, shape=(n, 2).
+
+    pol: string, optional
+        Polarization to use for bootstrapping. Default="xx".
+
+    n_boots: int, optional
+        Number of times to bootstrap to create a sample. Default: 50.
+    """
+    data = []
+    for uv in uvpl:
+        bspair = []
+        for u in uv:
+            uc = copy.deepcopy(u)
+            bspair.append(_bootstrap_uvp_once(uc, pol=pol, n_boots=n_boots))
+        data.append(bspair)
+    return data
+
+def save_jackknife(uvpl, savename=None):
+    """
+    Saves a bootstrapped jackknife pair list.
+
+    uvpl: list of UVPSpecs
+        The list to save. Note: will save jackknives with no bootstrap,
+        so make sure to run uvpl through bootstrap_jackknife first.
+
+    savename: string, optional
+        Outfile name. Will append with a bunch of information.
+        Default: None
+    """
+    if not os.path.exists("./data"):
+        os.mkdir("./data")
+
+    # Check if all jackknives are the same type
+    jkftypes = [u.jkftype  for uvp in uvpl for u in uvp]
+    if all([j == jkftypes[0] for j in jkftypes]):
+        jkftype = jkftypes[0]
+    else:
+        raise AttributeError("All jackknifes must be of the same type")
+
+    fname = "jackknife." + jkftype + ".Nj%i." % len(uvpl) + utils.timestamp()
+    if savename is not None:
+        fname = savename + "." + fname
+
+    # Open pspec container
+    pc = hp.PSpecContainer("./data/" + fname, "rw")
+
+    for i, uvp_pair in enumerate(uvpl):
+        jkf = "%s.%i" % (uvp_pair[0].jkftype, i)
+        for k, uvp in enumerate(uvp_pair):
+            # Save pspec to group {jackknife_type}.{jackknife_num} and 
+            # pspec grp{groupnumber}
+            name = "grp%i" % k
+            pc.set_pspec(jkf, psname=name, pspec=uvp)
+
+def split_ants(uvp, n_jacks=40, verbose=False):
```

Comment:
Maybe rename this to `split_ants_random`.

Reply:
Also, might be good to have a way of specifying the number of groups to make. 2 will be the default, but I can imagine people wanting more than 2 groups.

Reply:
What happens to the "group A" x "group B" power spectra?